tldw_Server_API/tests/Chat/unit/test_chat_service_tool_autoexec.py

```
@@ -188,6 +188,270 @@ async def save_message_fn(_db, _conv_id, payload, use_transaction=True):
+@pytest.mark.unit
+async def test_run_first_presented_tools_drive_autoexec_allow_catalog(monkeypatch: pytest.MonkeyPatch) -> None:
+    async def fake_log_llm_usage(**_kwargs):
+        return None
+
+    monkeypatch.setattr(chat_service, "log_llm_usage", fake_log_llm_usage)
+    monkeypatch.setattr(chat_service, "get_topic_monitoring_service", lambda: None)
+    monkeypatch.setattr(chat_service, "should_auto_execute_tools", lambda: True)
+    monkeypatch.setattr(chat_service, "get_chat_max_tool_calls", lambda: 2)
+    monkeypatch.setattr(chat_service, "get_chat_tool_timeout_ms", lambda: 3210)
+    monkeypatch.setattr(chat_service, "should_attach_tool_idempotency", lambda: False)
+    monkeypatch.setattr(chat_service, "resolve_chat_run_first_rollout_mode", lambda raw_mode=None, default="off": "gated")
```

Ruff (0.15.7)

[warning] Unused lambda argument: `raw_mode`

(ARG005)